File: Datasets/TrickyBugs/GenProgs/tc_generated_progs_cpp/p03239/p03239_num1_parsed.cpp
#include <stdio.h>

int N,T,c,t;
int C = 1001; // initialize C with a high value
bool Flag = false;

int main(void){
	scanf("%d %d",&N,&T);
	scanf("%d %d",&c,&t);
	if(t<=T) {
		Flag = true;
		C = c; // update C with the first route that takes not longer than time T
	}
	for(;N>1;N--){
		scanf("%d %d",&c,&t);
		if(t<=T && c<C){
			C=c; // update C with the smallest cost among the routes that takes not longer than time T
			Flag=true;
		}
	}
	if(Flag) printf("%d",C);
	else printf("TLE");
	return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03239/p03239_num1_parsed.py
# Here is the complete program to solve the given problem

# Read the input values
N, T = map(int, input().split())
routes = []
for _ in range(N):
    c, t = map(int, input().split())
    routes.append((c, t))

# Sort the routes based on their time in ascending order
routes.sort(key=lambda x: x[1])

# Find the smallest cost route that takes not longer than time T
min_cost = float('inf')
for c, t in routes:
    if t <= T:
        min_cost = min(min_cost, c)

# Print the result
if min_cost == float('inf'):
    print("TLE")
else:
    print(min_cost)